### ISSUE TITLE
The AutoAlias feature isn't applied from within the JOIN tree, only from the TableList

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14406]
Given these series:

```java
Table<Record1<Integer>> series = generateSeries(0, 3);
Table<Record1<Integer>> t = series.as("t", "a");
```

Both of these tests should work:

```java
assertEquals(3, create().fetchCount(
    selectOne()
    .from(t)
    .join(TBook())
    .on(TBook_ID().eq(t.field("a").coerce(Integer.class)))
));

assertEquals(3, create().fetchCount(
    selectOne()
    .from(series)
    .join(TBook())
    .on(TBook_ID().eq(series.field(0).coerce(Integer.class)))
));
```

But the second one fails, because the `AutoAlias` feature (https://github.com/jOOQ/jOOQ/issues/11564) doesn't work for `JOIN` trees. It only works for `TableList`.

The generated SQL for H2 is:

```sql
-- Explicit alias
select count(*) "c"
from (
  select 1 "v0"
  from system_range(0, 3) "t" ("a")
    join "PUBLIC"."T_BOOK"
      on "PUBLIC"."T_BOOK"."ID" = "t"."a"
) "t"

-- Missing auto alias
select count(*) "c"
from (
  select 1 "v0"
  from system_range(0, 3)
    join "PUBLIC"."T_BOOK"
      on "PUBLIC"."T_BOOK"."ID" = generate_series.generate_series
) "t"
```

For comparison, both of these work:

```java
assertEquals(3, create().fetchCount(
    selectOne()
    .from(t, TBook())
    .where(TBook_ID().eq(t.field("a").coerce(Integer.class)))
));

assertEquals(3, create().fetchCount(
    selectOne()
    .from(series, TBook())
    .where(TBook_ID().eq(series.field(0).coerce(Integer.class)))
));
```

With this generated SQL:

```sql
-- Explicit alias
select count(*) "c"
from (
  select 1 "v0"
  from
    system_range(0, 3) "t" ("a"),
    "PUBLIC"."T_BOOK"
  where "PUBLIC"."T_BOOK"."ID" = "t"."a"
) "t"

-- Correct auto alias
select count(*) "c"
from (
  select 1 "v0"
  from
    system_range(0, 3) generate_series (generate_series),
    "PUBLIC"."T_BOOK"
  where "PUBLIC"."T_BOOK"."ID" = generate_series.generate_series
) "t"
```

The problem doesn't appear in PostgreSQL for `GENERATE_SERIES`, because the `AutoAlias` corresponds to PostgreSQL's own automatic table/column names in this case.

**Comment by lukaseder** [comment]
The issue was discovered as part of https://github.com/jOOQ/jOOQ/issues/5799, and will be fixed along with the new feature. I'll check later if a backport is possible.

**Comment by lukaseder** [comment]
Given the complexity of the change related to #5799, I won't backport this particular fix. The fix will be committed with #5799.
